### PR TITLE
Literal table

### PR DESCRIPTION
**Pull request by shakram02**
- Remove ugly hex padding function

### CODE REVIEW

**Comment by shakram02** [comment]
meh :-1: forced update ruined the history

File: src/bin/main.rs [hunk 1]
```diff
@@ -39,17 +39,20 @@ fn main() {
...
-        let instruction = asm_file.read_instruction();
-        match instruction {
-            None => break,
-            Some(s) => {
-                println!("{:?}", s);
-            }
-        }
+
+    let asm_file = FileHandler::new(input);
+    let listing_info = sick_lib::pass_one::pass_one::pass_one(asm_file);
```
Comment (oddcoder) [inline src/bin/main.rs:44]: https://is.gd/ls0Oyu

File: Cargo.toml [hunk 2]
```diff
@@ -21,4 +21,5 @@ getopts = "*"
 regex = "*"
 log = "*"
 env_logger = "*"
-lazy_static = "*"
\ No newline at end of file
+lazy_static = "*"
+parking_lot = "*"
```
Comment (oddcoder) [inline Cargo.toml:25]: why ?

File: src/lib/basic_types/base_table.rs [hunk 3]
```diff
@@ -0,0 +1,105 @@
+/// When translator meets a base at a given
+/// location couter (xtr) it'll ask the state
+/// if a base is available for xtr
+
+use std::u32;
+use parking_lot::RwLock;
```
Comment (oddcoder) [inline src/lib/basic_types/base_table.rs:6]: we dont have multi threading here on any level so explain, mutexes or what ever you used here introduce new level of bugs and there should always be better ways to solve it, pls just dont over do it

File: src/lib/basic_types/formats.rs [hunk 4]
```diff
@@ -7,8 +7,13 @@ pub enum Format {
     Four = 4,
     None,
 }
-
+use std::fmt;
```
Comment (oddcoder) [inline src/lib/basic_types/formats.rs]: this should go to the top of the file

File: src/lib/htme/raw_program.rs [hunk 5]
```diff
@@ -5,11 +5,12 @@ use basic_types::instruction::Instruction;
 use basic_types::operands::*;
 use htme::record_string::*;
 
+#[derive(Debug)]
```
Comment (oddcoder) [inline src/lib/htme/raw_program.rs]: @Youssefares ^_^ ur part

File: src/lib/pass_one/pass_one.rs [hunk 6]
```diff
@@ -1,11 +1,16 @@
...
-use basic_types::instruction::*;
-use basic_types::formats::Format;
+use instruction::*;
+use formats::Format;
 use filehandler::*;
-use basic_types::unit_or_pair::*;
-use basic_types::operands::*;
+use unit_or_pair::*;
+use instruction_set::is_action_directive;
+use parking_lot::RwLock;
```
Comment (oddcoder) [inline src/lib/pass_one/pass_one.rs]: fuck mutexes ...

File: src/lib/pass_one/pass_one.rs [hunk 7]
```diff
@@ -41,43 +49,151 @@ fn get_instruction_size (inst:&Instruction) -> i32 {
...
-    let mut listing:Vec<Instruction> = Vec::new();
-    let mut instruction;
-    loop {
-        match file.read_instruction() {
-            None => break,
-            Some(inst) => instruction = inst,
-        }
+
+pub fn pass_one(mut file: FileHandler) -> (HashMap<String, i32>, RawProgram) {
+
```
Comment (oddcoder) [inline src/lib/pass_one/pass_one.rs:72]: remove all those empty lines ... they make readablity hard

File: src/lib/basic_types/base_table.rs (hunk 3 above)
Comment (shakram02) [reply src/lib/basic_types/base_table.rs:6]: static access needs locking

File: src/lib/pass_one/pass_one.rs (hunk 7 above)
Comment (shakram02) [reply src/lib/pass_one/pass_one.rs:72]: no they don't, it's a convention to leave empty lines in code and between functions

**Comment by shakram02** [comment]
this should also close #17

**Pull request merged**